collection-scripts/gather_performance_profile_controller

```diff
@@ -0,0 +1,162 @@
+do
+    node=$(echo $line | awk -F ' ' '{print $1}')
+    pod=$(echo $line | awk -F ' ' '{print $2}')
+    NODE_PATH=${NODES_PATH}/$node
+    mkdir -p "${NODE_PATH}"
+
+    oc exec $pod -n perf-node-gather -- lspci -nvv > $NODE_PATH/lspci
+    oc exec $pod -n perf-node-gather -- lscpu -e > $NODE_PATH/lscpu
+    oc exec $pod -n perf-node-gather -- cat /proc/cmdline > $NODE_PATH/proc_cmdline
+    oc exec $pod -n perf-node-gather -- dmesg > $NODE_PATH/dmesg
+    oc exec $pod -n perf-node-gather -- ethtool -k eth0 > $NODE_PATH/ethtool_features
+    oc exec $pod -n perf-node-gather -- ethtool -l eth0 > $NODE_PATH/ethtool_channels
+
+    COLLECTABLE_NODES+=($node)
+
+    oc exec $pod -n perf-node-gather -- gather_sysinfo --json cpuaff --procfs=/host/proc --sysfs=/host/sys > $NODE_PATH/cpu_affinities.json
+    oc exec $pod -n perf-node-gather -- gather_sysinfo --json irqaff --procfs=/host/proc --sysfs=/host/sys > $NODE_PATH/irq_affinities.json
+    oc exec $pod -n perf-node-gather -- gather_sysinfo --json podres --socket-path=unix:///host/podresources/kubelet.sock > $NODE_PATH/podresources.json
+
+    oc exec $pod -n perf-node-gather -- gather_sysinfo snapshot --debug --root=/host --output=- > $NODE_PATH/sysinfo.tgz 2> $NODE_PATH/sysinfo.log
+
+    oc exec $pod -n perf-node-gather -- gather_sysinfo podinfo --node-name $node > $NODE_PATH/pods_info.json
+done
+
+# Collect journal logs for specified units for all nodes
+NODE_UNITS=(kubelet)
+ADM_PIDS=()
+for NODE in ${COLLECTABLE_NODES[@]}; do
+    NODE_PATH=${NODES_PATH}/$NODE
+    mkdir -p ${NODE_PATH}
+    for UNIT in ${NODE_UNITS[@]}; do
+        timeout -k 5m 30m bash -c "oc adm node-logs $NODE -u $UNIT --since '-8h' | gzip" > ${NODE_PATH}/${NODE}_logs_$UNIT.gz &
+	ADM_PIDS+=($!)
+    done
+done
+wait "${ADM_PIDS[@]}"
+
+oc delete -f $DAEMONSET_MANIFEST
+oc delete -f $CLUSTER_ROLE_BINDING_MANIFEST
+oc delete -f $CLUSTER_ROLE_MANIFEST
+oc delete -f $SERVICEACCOUNT_MANIFEST
+oc delete -f $NAMESPACE_MANIFEST
+}
+
+function ppc_namespace() {
+  # we control the subs, so this is the most reliable way to get the namespace
+  ns=$( oc get subs -A --field-selector metadata.name='performance-addon-operator-subscription' -o=jsonpath='{.items[0].metadata.namespace}{"\n"}' 2> /dev/null )
+  # trying again with the pods, which are _usually_ reliable - but users can change them
+  [ -z "${ns}" ] && ns=$( oc get pods -A -l name='performance-operator' -o=jsonpath='{.items[0].metadata.namespace}{"\n"}' 2> /dev/null )
+  # namespace suggested by the documentation. This is a fancier way to check for its existence
+  [ -z "${ns}" ] && ns=$( oc get ns openshift-performance-addon-operator -o jsonpath='{.metadata.name}{"\n"}' 2> /dev/null )
+  # we should never get there. This is the last resort.
+  [ -z "${ns}" ] && ns="openshift-operators"
+  echo ${ns}
+}
+
+function version() {
```

Comment:
you shouldn't need to do this as every must-gather file includes a version file specifying exactly what was used as part of the collection.

Reply:
right